analyzers/src/SonarAnalyzer.CSharp/Rules/NotAssignedPrivateMember.cs

```diff
@@ -211,13 +211,22 @@ private static ISet<ISymbol> GetAssignedMemberSymbols(IList<MemberUsage> memberU
                 if (parentNode is ArgumentSyntax argument
                     && (!argument.RefOrOutKeyword.IsKind(SyntaxKind.None) || TupleExpressionSyntaxWrapper.IsInstance(argument.Parent)))
                 {
-                    assignedMembers.Add(memberSymbol);
+                    AddAssignedMember(assignedMembers, memberSymbol);
                 }
             }
 
             return assignedMembers;
         }
 
+        private static void AddAssignedMember(ISet<ISymbol> assignedMembers, ISymbol symbol)
+        {
+            assignedMembers.Add(symbol);
+            if (!symbol.Equals(symbol.OriginalDefinition))
+            {
+                assignedMembers.Add(symbol.OriginalDefinition);
+            }
```

Comment:
It's a `HashSet` so it will do the `Equals` check for us anyway under the hood. So we don't need to check it ourselves and we can directly just add it.
```suggestion
            assignedMembers.Add(symbol.OriginalDefinition);
```

Reply:
Having this as just two similar lines with no logic, we can make it a local function and remove the `assignedMembers` argument on the way.

Reply:
That was a debugging artifact :-)

Reply:
I prefer passing the arguments to the functions explicitly, rather than referencing something out of the scope :) but okay, will do it like this.